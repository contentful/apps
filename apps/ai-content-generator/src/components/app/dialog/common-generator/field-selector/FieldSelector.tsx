import useSupportedFields, { SupportedFieldTypes } from '@hooks/dialog/useSupportedFields';
import { Flex } from '@contentful/f36-components';
import { GeneratorAction, GeneratorParameters } from '../generatorReducer';
import { ChangeEvent, useContext, useEffect } from 'react';
import EntryFieldList from './field-list/EntryFieldList';
import { GeneratorContext } from '@providers/generatorProvider';
import {
  getFieldData,
  updateOutputField,
  updateSourceField,
} from '@utils/dialog/common-generator/field-selector/fieldSelectorHelpers';

interface Props {
  parameters: GeneratorParameters;
  fieldTypes: SupportedFieldTypes[];
}

const FieldSelector = (props: Props) => {
  const { parameters, fieldTypes } = props;
  const { isNewText, sourceField, output } = parameters;
  const { entryId, dispatch, fieldLocales, localeNames, defaultLocale } =
    useContext(GeneratorContext);

  const { supportedFieldsWithContent, allSupportedFields } = useSupportedFields(
    entryId,
    fieldTypes,
    fieldLocales,
    localeNames,
    defaultLocale
  );

  const handleSelectChange = (action: GeneratorAction) => {
    if (action === GeneratorAction.UPDATE_SOURCE_FIELD) {
      return (event: ChangeEvent<HTMLSelectElement>) => {
        const sourceFieldData = getFieldData(event.target.value, supportedFieldsWithContent);
        updateSourceField(sourceFieldData, supportedFieldsWithContent[0], dispatch);
      };
    }

    if (action === GeneratorAction.UPDATE_OUTPUT_FIELD) {
      return (event: ChangeEvent<HTMLSelectElement>) => {
        const outputFieldData = getFieldData(event.target.value, allSupportedFields);
        updateOutputField(outputFieldData, allSupportedFields[0], dispatch);
      };
    }

    return (event: ChangeEvent<HTMLSelectElement>) =>
      dispatch({
        type: action,
        value: event.target.value,
      });
  };

  const handleBaseDataChange = () => {
    if (supportedFieldsWithContent.length) {
      const sourceFieldData = getFieldData(
        isNewText ? sourceField : '',
        supportedFieldsWithContent
      );
      updateSourceField(sourceFieldData, supportedFieldsWithContent[0], dispatch);
    }

    if (allSupportedFields.length) {
      const outputFieldData = getFieldData(output.fieldId, allSupportedFields);
      updateOutputField(outputFieldData, allSupportedFields[0], dispatch);
    }
  };

  useEffect(handleBaseDataChange, [isNewText]);

  return (
    <Flex flexGrow={2} flexDirection="column" margin="spacingL">
      {!isNewText && (
        <EntryFieldList
          title="Source Field"
          selectedField={sourceField}
          fields={supportedFieldsWithContent}
          onChange={handleSelectChange(GeneratorAction.UPDATE_SOURCE_FIELD)}
        />
      )}

      <EntryFieldList
        title="Output Field"
        selectedField={output.fieldKey}
        fields={allSupportedFields}
        onChange={handleSelectChange(GeneratorAction.UPDATE_OUTPUT_FIELD)}
      />
    </Flex>
  );
};

export default FieldSelector;
