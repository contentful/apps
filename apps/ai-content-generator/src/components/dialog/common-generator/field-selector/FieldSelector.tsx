import useSupportedFields, { SupportedFieldTypes } from '@hooks/dialog/useSupportedFields';
import { Flex } from '@contentful/f36-components';
import { GeneratorAction, GeneratorReducer, GeneratorParameters } from '../generatorReducer';
import { Dispatch, useEffect } from 'react';
import AvailableLocales from './available-locales/AvailableLocales';
import EntryFieldList from './field-list/EntryFieldList';

interface Props {
  parameters: GeneratorParameters;
  entryId: string;
  isTranslate?: boolean;
  fieldTypes: SupportedFieldTypes[];
  dispatch: Dispatch<GeneratorReducer>;
}

const FieldSelector = (props: Props) => {
  const { parameters, entryId, isTranslate, fieldTypes, dispatch } = props;
  const { isNewText, sourceField, outputField, locale, targetLocale } = parameters;
  const { supportedFields, fields } = useSupportedFields(entryId, fieldTypes, parameters.locale);

  useEffect(() => {
    if (fields.length) {
      dispatch({
        type: GeneratorAction.SOURCE_FIELD,
        field: fields[0].name,
        value: parameters.isNewText ? '' : fields[0].data,
      });
    }

    if (supportedFields.length) {
      dispatch({
        type: GeneratorAction.OUTPUT_FIELD,
        value: supportedFields[0].name,
      });
    }
  }, [fields, supportedFields, parameters.isNewText]);

  const handleSelectChange = (eventType: GeneratorAction) => {
    return (event: React.ChangeEvent<HTMLSelectElement>) => {
      const sourceFieldName = event.target.value;

      if (eventType === GeneratorAction.SOURCE_FIELD) {
        const newOriginalText = fields.find((field) => field.name === sourceFieldName);

        dispatch({
          type: GeneratorAction.SOURCE_FIELD,
          field: sourceFieldName,
          value: newOriginalText?.data || '',
        });
      } else {
        dispatch({
          type: eventType,
          value: sourceFieldName,
        });
      }
    };
  };

  const { SOURCE_FIELD, OUTPUT_FIELD, LOCALE, TARGET_LOCALE } = GeneratorAction;
  return (
    <Flex flexGrow={2} flexDirection="column" margin="spacingL">
      {!isNewText && (
        <EntryFieldList
          title="Source Field"
          selectedField={sourceField}
          fields={fields}
          onChange={handleSelectChange(SOURCE_FIELD)}
        />
      )}

      <EntryFieldList
        title="Output Field"
        selectedField={outputField}
        fields={supportedFields}
        onChange={handleSelectChange(OUTPUT_FIELD)}
      />

      <AvailableLocales
        title={isTranslate ? 'Source Language' : 'Language'}
        selectedLocale={locale}
        onChange={handleSelectChange(LOCALE)}
      />

      {isTranslate && (
        <AvailableLocales
          title="Target Language"
          selectedLocale={targetLocale || locale}
          onChange={handleSelectChange(TARGET_LOCALE)}
        />
      )}
    </Flex>
  );
};

export default FieldSelector;
