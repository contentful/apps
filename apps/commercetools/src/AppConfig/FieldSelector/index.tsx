import * as React from 'react';
import get from 'lodash/get';
import set from 'lodash/set';
import omit from 'lodash/omit';
import tokens from '@contentful/forma-36-tokens';
import { css } from 'emotion';
import { Form, Subheading, CheckboxField } from '@contentful/forma-36-react-components';

import { ToggleGroup } from './ToggleGroup';
import { ContentType, CompatibleFields, FieldsConfig } from '../fields';
import { PickerMode } from '../../interfaces';

const styles = {
  fieldGroup: css({
    display: 'flex',
    flexDirection: 'column'
  }),
  select: css({
    marginLeft: '10px'
  })
};

interface Props {
  contentTypes: ContentType[];
  compatibleFields: CompatibleFields;
  selectedFields: FieldsConfig;
  onSelectedFieldsChange: Function;
}

export default class FieldSelector extends React.Component<Props> {
  onSelectedFieldChange = (
    ctId: string,
    fieldId: string,
    type: string,
    e: React.ChangeEvent<HTMLInputElement>
  ) => {
    const isTargetChecked = e.currentTarget.checked;

    this.props.onSelectedFieldsChange(
      isTargetChecked
        ? set(Object.assign({}, this.props.selectedFields), [ctId, fieldId], type)
        : omit(this.props.selectedFields, [`${ctId}.${fieldId}`])
    );
  };

  onSelectedFieldTypeChange(ctId: string, fieldId: string, type: PickerMode) {
    this.props.onSelectedFieldsChange(
      set(Object.assign({}, this.props.selectedFields), [ctId, fieldId], type)
    );
  }

  render() {
    const { compatibleFields, contentTypes, selectedFields } = this.props;

    return contentTypes.map(ct => {
      const fields = compatibleFields[ct.sys.id];
      return (
        <div key={ct.sys.id} className={css({ marginTop: tokens.spacingL })}>
          <Subheading>{ct.name}</Subheading>
          <Form>
            {fields.map(field => {
              const type = get(selectedFields, [ct.sys.id, field.id], null);
              const isChecked = !!type;

              return (
                <div className={styles.fieldGroup} key={field.id}>
                  <CheckboxField
                    id={`field-box-${ct.sys.id}-${field.id}`}
                    labelText={field.name}
                    helpText={`Short text Â· Field ID: ${field.id}`}
                    checked={isChecked}
                    onChange={this.onSelectedFieldChange.bind(this, ct.sys.id, field.id, 'product')}
                  />
                  {isChecked && (
                    <ToggleGroup
                      activePickerMode={type}
                      onChange={type => this.onSelectedFieldTypeChange(ct.sys.id, field.id, type)}
                    />
                  )}
                </div>
              );
            })}
          </Form>
        </div>
      );
    });
  }
}
