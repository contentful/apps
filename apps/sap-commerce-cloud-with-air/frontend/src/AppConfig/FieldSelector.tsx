import * as React from 'react';
import tokens from '@contentful/forma-36-tokens';
import { css } from '@emotion/css';
import { Checkbox, Flex, Form, Subheading } from '@contentful/f36-components';
import { ContentType, CompatibleFields, SelectedFields } from './fields';

interface Props {
  contentTypes: ContentType[];
  compatibleFields: CompatibleFields;
  selectedFields: SelectedFields;
  onSelectedFieldsChange: Function;
}
export default class FieldSelector extends React.Component<Props> {
  onSelectedFieldChange = (
    ctId: string,
    fieldId: string,
    e: React.ChangeEvent<HTMLInputElement>,
  ) => {
    const updated = { ...this.props.selectedFields };

    if (e.currentTarget.checked) {
      updated[ctId] = (updated[ctId] || []).concat([fieldId]);
    } else {
      updated[ctId] = (updated[ctId] || []).filter((cur) => cur !== fieldId);
    }

    this.props.onSelectedFieldsChange(updated);
  };

  render() {
    const { compatibleFields, contentTypes, selectedFields } = this.props;

    return (
      <>
        {contentTypes.map((ct) => {
          const fields = compatibleFields[ct.sys.id];
          return (
            <div key={ct.sys.id} className={css({ marginTop: tokens.spacingL })}>
              <Subheading style={{ marginBottom: tokens.spacingM }}>{ct.name}</Subheading>
              <Form>
                {fields.map((field) => (
                  <Flex marginTop="spacingM" flexDirection="column" gap="spacingXs" key={field.id}>
                    <Checkbox
                      id={`field-box-${ct.sys.id}-${field.id}`}
                      helpText={`${
                        field.type === 'Symbol' ? 'Short text' : 'Short text, list'
                      } Â· Field ID: ${field.id}`}
                      isChecked={(selectedFields[ct.sys.id] || []).includes(field.id)}
                      onChange={this.onSelectedFieldChange.bind(this, ct.sys.id, field.id)}>
                      {field.name}
                    </Checkbox>
                  </Flex>
                ))}
              </Form>
            </div>
          );
        })}
      </>
    );
  }
}
