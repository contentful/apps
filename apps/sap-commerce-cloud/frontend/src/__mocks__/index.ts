export { makeSdkMock } from './mockSdk';
export { mockProductPreview } from './mockProductPreview';
