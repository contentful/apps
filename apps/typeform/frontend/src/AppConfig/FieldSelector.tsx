import * as React from 'react';
import tokens from '@contentful/f36-tokens';
import { css } from '@emotion/css';
import { Form, FormControl, Subheading, Checkbox, Typography } from '@contentful/f36-components';

import { ContentType, CompatibleFields, SelectedFields } from '../typings';

export interface Props {
  contentTypes: ContentType[];
  compatibleFields: CompatibleFields;
  selectedFields: SelectedFields;
  onSelectedFieldsChange: Function;
}

export default class FieldSelector extends React.Component<Props> {
  onSelectedFieldChange = (
    ctId: string,
    fieldId: string,
    e: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updated = { ...this.props.selectedFields };

    if (e.currentTarget.checked) {
      updated[ctId] = (updated[ctId] || []).concat([fieldId]);
    } else {
      updated[ctId] = (updated[ctId] || []).filter((cur) => cur !== fieldId);
    }

    this.props.onSelectedFieldsChange(updated);
  };

  render() {
    const { compatibleFields, contentTypes, selectedFields } = this.props;

    return (
      <>
        {contentTypes.map((ct) => {
          const fields = compatibleFields[ct.sys.id];
          return (
            <div key={ct.sys.id} className={css({ marginTop: tokens.spacingL })}>
              <Subheading>{ct.name}</Subheading>
              <Form>
                {fields.map((field) => (
                  <FormControl key={field.id} id={`field-box-${ct.sys.id}-${field.id}`}>
                    <Checkbox
                      helpText={`Field ID: ${field.id}`}
                      isChecked={(selectedFields[ct.sys.id] || []).includes(field.id)}
                      onChange={this.onSelectedFieldChange.bind(this, ct.sys.id, field.id)}>
                      {field.name}
                    </Checkbox>
                  </FormControl>
                ))}
              </Form>
            </div>
          );
        })}
      </>
    );
  }
}
