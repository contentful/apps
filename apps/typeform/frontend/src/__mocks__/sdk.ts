export const sdk: any = {
  ids: {
    space: 'test-space',
    environment: 'master',
    user: '123',
  },
  app: {
    setReady: jest.fn(),
    isInstalled: jest.fn().mockReturnValue(Promise.resolve(false)),
    getParameters: jest.fn().mockReturnValue(Promise.resolve(null)),
    onConfigure: jest.fn(),
  },
  space: {
    getContentTypes: jest.fn().mockReturnValue(Promise.resolve([])),
    getEditorInterfaces: jest.fn().mockResolvedValue({ items: [] }),
  },
  notifier: {
    error: jest.fn(),
  },
  window: {
    startAutoResizer: jest.fn(),
  },
  field: {
    getValue: jest.fn().mockReturnValue('field-value'),
    setValue: jest.fn(),
    removeValue: jest.fn(),
  },
  parameters: {
    installation: {
      workspaceId: 'mock-workspace-id',
      accessToken: 'mock-access-token',
    },
  },
};
