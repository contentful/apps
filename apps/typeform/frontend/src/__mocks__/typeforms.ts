export const typeforms = {
  forms: {
    total_items: 2,
    page_count: 1,
    items: [
      {
        id: 'OhBix6',
        title: 'Customer Feedback Survey (copy)',
        last_updated_at: '2020-03-03T14:07:00Z',
        settings: { is_public: false, is_trial: false },
        self: { href: 'https://api.typeform.com/forms/Oksaodk' },
        theme: { href: 'https://api.typeform.com/themes/fkoaskfd' },
        _links: { display: 'https://user.typeform.com/to/dsako' },
      },
      {
        id: 'sUM1zW',
        title: 'Personality Quiz - Quick Start - FINAL (copy)',
        last_updated_at: '2020-03-03T14:05:55Z',
        settings: { is_public: true, is_trial: true },
        self: { href: 'https://api.typeform.com/forms/dsjao' },
        theme: { href: 'https://api.typeform.com/themes/dsajo' },
        _links: { display: 'https://user.typeform.com/to/sjao' },
      },
    ],
  },
};
