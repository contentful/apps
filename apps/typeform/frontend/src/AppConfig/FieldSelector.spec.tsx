import React from 'react';
import { configure, render, cleanup } from '@testing-library/react';
import FieldSelector, { Props } from './FieldSelector';
import { typeforms } from '../__mocks__/typeforms';
import { sdk as mockSdk } from '../__mocks__/sdk';

configure({
  testIdAttribute: 'data-test-id'
});

window.fetch = jest.fn(() => ({
  json: () => typeforms
})) as any;

const defaultProps: Props = {
  contentTypes: [
    {
      sys: { id: 'ct1' },
      name: 'CT1',
      fields: [
        { id: 'x', name: 'X', type: 'Symbol' },
        { id: 'y', name: 'Y', type: 'Object' }
      ]
    },
    {
      sys: { id: 'ct2' },
      name: 'CT2',
      fields: [
        { id: 'foo', name: 'FOO', type: 'Text' },
        { id: 'z', name: 'Z', type: 'Array', items: { type: 'Symbol' } }
      ]
    }
  ],
  compatibleFields: {
    ct1: [{ id: 'x', name: 'X', type: 'Symbol' }],
    ct2: []
  },
  selectedFields: {},
  onSelectedFieldsChange: jest.fn()
};

describe('FieldSelector', () => {
  afterEach(cleanup);

  it('should render successfully with no preselected fields', async () => {
    const component = render(<FieldSelector {...defaultProps} />);
    expect(component.container).toMatchSnapshot();
  });

  it('should render successfully with preselected fields', async () => {
    const props = {
      ...defaultProps,
      selectedFields: { ct1: ['x'] }
    };
    const component = render(<FieldSelector {...props} />);
    expect(component.container).toMatchSnapshot();
  });
});
