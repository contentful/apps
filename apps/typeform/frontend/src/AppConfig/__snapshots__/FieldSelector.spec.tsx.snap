// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldSelector should render successfully with no preselected fields 1`] = `
<div>
  <div
    class="Typography__Typography___1ZUfE"
    data-test-id="cf-ui-text-container"
  >
    <div
      class="css-1w2p18d"
    >
      <h2
        class="Subheading__Subheading___2mA9j f36-margin-bottom--m"
        data-test-id="cf-ui-subheading"
      >
        CT1
      </h2>
      <form
        class="Form__Form___2zwZf"
        data-test-id="cf-ui-form"
      >
        <div
          class="Form__Form__item___2qOZE Form__Form__item--default___1CHMK"
        >
          <div
            class="ControlledInputField__ControlledInputField___2uIG9"
            data-test-id="cf-ui-checkbox-field"
          >
            <div
              class="ControlledInput__Checkbox___ra2TY ControlledInputField__ControlledInputField__input___3OMYB"
            >
              <input
                aria-label="X"
                class="ControlledInput__ControlledInput___2XK3j"
                data-test-id="cf-ui-controlled-input"
                id="field-box-ct1-x"
                type="checkbox"
                value=""
              />
              <label
                class="ControlledInput__Input__ghost___2bBB9 ControlledInput__Checkbox__ghost___1Oqlk"
                for="field-box-ct1-x"
              >
                <svg
                  class="Icon__Icon___38Epv Icon__Icon--medium___3Pezi Icon__Icon--white___3GVPJ"
                  data-test-id="cf-ui-icon"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                  />
                </svg>
              </label>
            </div>
            <div>
              <label
                class="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
                data-test-id="cf-ui-form-label"
                for="field-box-ct1-x"
              >
                X
              </label>
              <p
                class="HelpText__HelpText___uWbja"
                data-test-id="cf-ui-help-text"
              >
                Field ID: x
              </p>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="css-1w2p18d"
    >
      <h2
        class="Subheading__Subheading___2mA9j f36-margin-bottom--m"
        data-test-id="cf-ui-subheading"
      >
        CT2
      </h2>
      <form
        class="Form__Form___2zwZf"
        data-test-id="cf-ui-form"
      />
    </div>
  </div>
</div>
`;

exports[`FieldSelector should render successfully with preselected fields 1`] = `
<div>
  <div
    class="Typography__Typography___1ZUfE"
    data-test-id="cf-ui-text-container"
  >
    <div
      class="css-1w2p18d"
    >
      <h2
        class="Subheading__Subheading___2mA9j f36-margin-bottom--m"
        data-test-id="cf-ui-subheading"
      >
        CT1
      </h2>
      <form
        class="Form__Form___2zwZf"
        data-test-id="cf-ui-form"
      >
        <div
          class="Form__Form__item___2qOZE Form__Form__item--default___1CHMK"
        >
          <div
            class="ControlledInputField__ControlledInputField___2uIG9"
            data-test-id="cf-ui-checkbox-field"
          >
            <div
              class="ControlledInput__Checkbox___ra2TY ControlledInputField__ControlledInputField__input___3OMYB"
            >
              <input
                aria-label="X"
                checked=""
                class="ControlledInput__ControlledInput___2XK3j"
                data-test-id="cf-ui-controlled-input"
                id="field-box-ct1-x"
                type="checkbox"
                value=""
              />
              <label
                class="ControlledInput__Input__ghost___2bBB9 ControlledInput__Checkbox__ghost___1Oqlk"
                for="field-box-ct1-x"
              >
                <svg
                  class="Icon__Icon___38Epv Icon__Icon--medium___3Pezi Icon__Icon--white___3GVPJ"
                  data-test-id="cf-ui-icon"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                  />
                </svg>
              </label>
            </div>
            <div>
              <label
                class="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
                data-test-id="cf-ui-form-label"
                for="field-box-ct1-x"
              >
                X
              </label>
              <p
                class="HelpText__HelpText___uWbja"
                data-test-id="cf-ui-help-text"
              >
                Field ID: x
              </p>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="css-1w2p18d"
    >
      <h2
        class="Subheading__Subheading___2mA9j f36-margin-bottom--m"
        data-test-id="cf-ui-subheading"
      >
        CT2
      </h2>
      <form
        class="Form__Form___2zwZf"
        data-test-id="cf-ui-form"
      />
    </div>
  </div>
</div>
`;
